### PR TITLE
updated the flake autoprioritization to reflect histogram

### LINKED ISSUE #1686: Update The Auto-prioritization of Flakes
p0 Flakes are currently defined as 3x per week.  That results in a lot of p0 flakes.  Let's recalibrate

### PR DESCRIPTION
**Pull request by grodrigues3**
Fixes #1686

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1687)

### CODE REVIEW

**Comment by grodrigues3** [comment]
flake_graphics

**Comment by freehan** [comment]
Do everyone agree on that  >50 is P1 and >80 is P0? 

I feel that is too relaxed. Only test in Node E2E can reach that.

**Comment by lavalamp** [comment]
Goal is to fight priority inflation, so a P0 means something again.

@pwittrock @matchstick

**Comment by test-foxish** [comment]
recomputing cla status...

**Comment by grodrigues3** [comment]
I signed it!

**Comment by lavalamp** [comment]
@dchen1107 We were talking about this earlier today

**Comment by lavalamp** [comment]
I would like to get this in. I'd also like a change (perhaps run weekly?) that downgrades current priorities if the issues don't meet the bar.

**Comment by lavalamp** [comment]
@grodrigues3 maybe you could re-gen your chart, just to see if current data looks similar or not.

**Comment by grodrigues3** [comment]
@apelisse Can you pull the data from your SQL db and I'll bin and plot it

**Comment by grodrigues3** [comment]
@lavalamp here's a histogram reflecting the test flakes as of Thursday 10/05/2016.

What do you think?


test_flake_dist1005

**Comment by lavalamp** [comment]
I would say, the first bar there is p3, the second bar is p2, the third bar
is p1, and everything else is p0. P0 should be like 95th percentile...

**Comment by grodrigues3** [comment]
Some stats on the data.  95th percentile seems to be about 50 flakes/week.

The Range of the Data is 1 - 1247
25 Percentile: 1.0
50 Percentile: 3.0
75 Percentile: 10.0
95 Percentile: 51.4

I'll update the pr to reflect this.

**Comment by lavalamp** [comment]
Those numbers look like good p3-p0 boundaries to me. Thanks!

**Comment by grodrigues3** [comment]
@lavalamp made the tests pass too, can you PTAL and lgtm when it looks good

File: mungegithub/mungers/flake-manager.go
Comment (apelisse) [inline mungegithub/mungers/flake-manager.go]: Thanks for keeping the comment up-to-date

**Review by apelisse — APPROVED** [review]

**Comment by apelisse** [comment]
I'll let @lavalamp have the final word

**Comment by grodrigues3** [comment]
@foxish something weird happened with the CNCF bot.  Looks like it added and removed CNCF label even though I signed that a few weeks ago.

**Comment by foxish** [comment]
@grodrigues3 That is very odd indeed. I'll check with the CNCF if they are seeing issues.

**Pull request merged**
